src/extent.c

```diff
@@ -319,6 +319,8 @@ extents_insert_locked(tsdn_t *tsdn, extents_t *extents, extent_t *extent,
 		    (size_t)pind);
 	}
 	extent_heap_insert(&extents->heaps[pind], extent);
+	extent_t *first_extent = extent_heap_first(&extents->heaps[pind]);
```

Comment:
The current pairing heap code currently does a merge for the first() operation, changing the complexity here from O(1) to O(log n).  Can we just do a comparison against the current lowest_sn_heaps instead? (or change the pairing heap code...)